mtp_api/apps/transaction/permissions.py

```diff
@@ -1,11 +1,29 @@
 from rest_framework.permissions import BasePermission
 
+from core.permissions import ActionsBasedPermissions
+
+
 class IsOwner(BasePermission):
 
     def has_permission(self, request, view):
         return str(request.user.id) == view.kwargs.get('user_id')
 
+
 class IsOwnPrison(BasePermission):
 
     def has_permission(self, request, view):
         return request.user.prisonusermapping.prisons.filter(pk=view.kwargs.get('prison_id')).exists()
+
+
+class TransactionPermissions(ActionsBasedPermissions):
+    actions_perms_map = {
+        'create': ['%(app_label)s.add_%(model_name)s'],
+        'list': ['%(app_label)s.view_%(model_name)s'],
+        'retrieve': ['%(app_label)s.view_%(model_name)s'],
+        'update': ['%(app_label)s.change_%(model_name)s'],
+        'partial_update': ['%(app_label)s.change_%(model_name)s'],
+        'destroy': ['%(app_label)s.delete_%(model_name)s'],
+        'take': ['%(app_label)s.take_%(model_name)s'],
+        'release': ['%(app_label)s.release_%(model_name)s'],
+        'patch_credited': ['%(app_label)s.patch_credited_%(model_name)s']
+    }
```

Comment:
Could also do it in an ```__init__``` override.

```python
        def __init__(self, *args, **kwargs):
                self.action_perms_map.update({...})
                super(ActionBasedPermissions, self).__init__(*args, **kwargs)
```

That feels a bit dodgy, though, and could produce unintended behaviour if e.g. someone subclasses a subclass with an update in the init method and then also redefines the attribute in an attempt to override completely.